src/mango.ts

```diff
@@ -0,0 +1,493 @@
+      console.log(scalper.symbol, 'Canceling All Orders');
+      await mangoClient.perpCancelAllOrders(
+        mangoGroup,
+        mangoAccount,
+        perpMarket.perpMarketIndex,
+        10,
+      );
+      break;
+    }
+  }
+}
+
+export async function deltaHedgeMango(
+  dipProduct: DIPDeposit[],
+  scalper: Scalper,
+  mangoClient: MangoClient,
+  mangoAccount: MangoAccount,
+  mangoGroup: Group,
+  perpMarket: PerpMarket,
+  spotMarket: Market,
+  fillFeed: WebSocket,
+  deltaHedgeCount: number,
+): Promise<void> {
+  await mangoAccount.reload(mangoClient);
+  await mangoGroup.reloadAll(mangoClient);
+  // Cleanup from previous runs.
+  await cancelStaleMangoOrders(scalper, mangoClient, mangoAccount, mangoGroup, perpMarket);
+
+  // Avoid unsafe recursion.
+  if (deltaHedgeCount > MAX_DELTA_HEDGES) {
+    console.log(scalper.symbol, 'Max Hedges exceeded without getting to neutral');
+    return;
+  }
+
+  // Calc DIP delta for new position
+  const fairValue = perpMarket.uiPrice;
+  const dipTotalDelta = getDIPDelta(dipProduct, fairValue, scalper.symbol);
+
+  // Get Mango delta position
+  const mangoPerpDelta = mangoAccount.getPerpPositionUi(mangoGroup, perpMarket.perpMarketIndex);
+  let mangoSpotDelta = 0;
+  try {
+    mangoSpotDelta = mangoAccount.getTokenBalanceUi(
+      mangoGroup.getFirstBankByMint(tokenToSplMint(scalper.symbol)),
+    );
+  } catch (err) {
+    console.log(scalper.symbol, 'No Mango Token Balance Possible');
+  }
+
+  // Get all spot positions Option Vault, Risk Manager, Mango Tester
+  const spotDelta = await getSpotDelta(
+    scalper.connection,
+    scalper.symbol,
+    scalper.owner,
+    spotMarket,
+  );
+
+  // Get Total Delta Position to hedge
+  let hedgeDeltaTotal = IS_DEV
+    ? 0.1
+    : mangoPerpDelta + dipTotalDelta + spotDelta + mangoSpotDelta + scalper.deltaOffset;
+
+  // Check whether we need to hedge.
+  const dipTotalGamma = getDIPGamma(dipProduct, fairValue, scalper.symbol);
+  const stdDevSpread = (scalper.impliedVol / Math.sqrt(SEC_PER_YEAR
+    / SCALPER_WINDOW_SEC)) * scalper.zScore;
+  const deltaThreshold = Math.max(
+    dipTotalGamma * stdDevSpread * fairValue * GAMMA_THRESHOLD,
+    scalper.minSize,
+  );
+  if (Math.abs(hedgeDeltaTotal) < (deltaThreshold)) {
+    console.log(scalper.symbol, 'Delta Netural', hedgeDeltaTotal, '<', deltaThreshold);
+    return;
+  }
+
+  // TODO: Add slippage delta adjustment
+  const slippageTolerance = Math.min(stdDevSpread / 2, slippageMax.get(scalper.symbol));
+  const IS_BUYSIDE = hedgeDeltaTotal < 0;
+  let hedgePrice = IS_BUYSIDE
+    ? fairValue * (1 + slippageTolerance)
+    : fairValue * (1 - slippageTolerance);
+  const slippageDIPDelta = getDIPDelta(dipProduct, hedgePrice, scalper.symbol);
+  const dipDeltaDiff = slippageDIPDelta - dipTotalDelta;
+  hedgeDeltaTotal += dipDeltaDiff;
+  console.log(scalper.symbol, 'Adjust Slippage Delta by', dipDeltaDiff, 'to', -hedgeDeltaTotal);
```

Comment:
move the log line to the end of the logical block or just add a blank line after this to be similar to the pattern elsewhere

// comment on what the calculations are for
code
code
code
console.log('The results of that computation')

// comment on what the calculations are for
code
code
code
console.log('The results of that computation')

Reply:
Overhalued this section. Condensed logs